src/mono/mono/metadata/assembly.c

```diff
@@ -2593,6 +2574,27 @@ open_from_bundle_internal (MonoAssemblyLoadContext *alc, const char *filename, M
 	return image;
 }
 
+static MonoImage *
+open_from_satellite_bundle (MonoAssemblyLoadContext *alc, const char *filename, MonoImageOpenStatus *status, gboolean refonly, const char *culture)
+{
+	if (!satellite_bundles)
+		return NULL;
+
+	MonoImage *image = NULL;
+	char *name = g_strdup (filename);
+	for (int i = 0; !image && satellite_bundles [i]; ++i) {
+		if (strcmp (satellite_bundles [i]->name, name) == 0 && strcmp (satellite_bundles [i]->culture, culture) == 0) {
+			char *bundle_name = g_strconcat (name, "/", culture, (const char *)NULL);
```

Comment:
`"xyz.dll.resources/es-ES"` ?  curious why you swapped the usual order.

also maybe pull the naming out as a separate utility funtion?